src/utils/handleTPARedirect.js

```diff
@@ -0,0 +1,18 @@
+import { getConfig } from '@edx/frontend-platform';
+
+
+const handleTPARedirect = (TPAQueryparam) => {
+
+  // Get the URL of the login page in the other project
+  const loginUrl = `${getConfig().LOGIN_URL}`;
+
+  // Set the URL parameters for the redirect back to your home page after login
+  const redirectParams = `?next=${encodeURIComponent(
+    document.referrer
+  )}&origin=${window.location.origin}`;
+
+  // Redirect the user to the login page with the redirect parameters
+  window.location.href = `${loginUrl}&${TPAQueryparam}&${redirectParams}`;
```

Comment:
isn't better to use react.rounter for redirecting? as it provides a page-reload prevention. but if page reload is intentional in this case just ignore this comment

Reply:
Honestly, I followed the logic used before for handling the the login button, with the bare functionality in mind. Feel free to do a refactor on it.